### ISSUE TITLE
react-router v5 compatibility

### ISSUE THREAD

**Issue by jwineman** [jcoreio/react-router-transition-switch#1]
I tried upgrading to `react-router` v5 and got the following errors:
```
Warning: Please use `require("react-router").matchPath` instead of `require("react-router/matchPath")`. Support for the latter will be removed in the next major release.
```
and
```
You should not use <TransitionSwitch> outside a <Router>
Invariant Violation: You should not use <TransitionSwitch> outside a <Router>
    at invariant (/Users/johnwineman/projects/front-end/node_modules/invariant/invariant.js:40:15)
    at TransitionSwitch.componentWillMount (/Users/johnwineman/projects/front-end/node_modules/react-router-transition-switch/lib/index.js:62:31)
    at processChild (/Users/johnwineman/projects/front-end/node_modules/react-dom/cjs/react-dom-server.node.development.js:2905:16)
    at resolve (/Users/johnwineman/projects/front-end/node_modules/react-dom/cjs/react-dom-server.node.development.js:2795:5)
    at ReactDOMServerRenderer.render (/Users/johnwineman/projects/front-end/node_modules/react-dom/cjs/react-dom-server.node.development.js:3185:22)
    at ReactDOMServerRenderer.read (/Users/johnwineman/projects/front-end/node_modules/react-dom/cjs/react-dom-server.node.development.js:3144:29)
    at renderToString (/Users/johnwineman/projects/front-end/node_modules/react-dom/cjs/react-dom-server.node.development.js:3629:27)
    at request (/Users/johnwineman/projects/front-end/dist/server/map/webpack:/src/server/middleware/render.js:43:20)
    at /Users/johnwineman/projects/front-end/dist/server/map/webpack:/src/util/server.util.js:25:23
    at <anonymous>
```

This is because your plugin is checking for the router context using the legacy API.  I filed https://github.com/ReactTraining/react-router/issues/6656 to see how we should detect the `<Router/>` in the new version.

**Comment by jedwards1211** [comment]
Thanks for reporting this.  Maybe we should ask them to make `matchPath` an officially supported API as well?

**Comment by jedwards1211** [comment]
Nevermind, I guess it is already public API

**Comment by jedwards1211** [comment]
okay, if the CI build passes it should release a fix for this!

**Comment by jwineman** [comment]
Nice! I don't see a PR, curious on your approach - how did you fix?

**Comment by jwineman** [comment]
Nvm I see https://github.com/jcoreio/react-router-transition-switch/commit/be8f9fc7b604331f4f2cb902a2d2036f83c92c8d

**Comment by jedwards1211** [comment]
Yeah I used `withRouter`, it was a little awkward dealing with `location` overrides but better than asking them to make `RouterContext` public API :smile:

**Comment by jedwards1211** [comment]
Okay, fix published in [v1.1.1](https://github.com/jcoreio/react-router-transition-switch/releases/tag/v1.1.1)!

**Comment by jedwards1211** [comment]
> Appreciate the quick follow up by the way!

No problem, sorry for the friction -- I regret using their internal context API in the first place (I had used their `Switch` code as a starting point)
